tests/end-to-end/director/runtime_auth_test.go

```diff
@@ -0,0 +1,149 @@
+package director
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+
+	gcli "github.com/machinebox/graphql"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestQueryRuntimeAuths(t *testing.T) {
+	// GIVEN
+	ctx := context.Background()
+
+	defaultAuth := fixBasicAuth()
+	customAuth := graphql.AuthInput{
+		Credential: &graphql.CredentialDataInput{
+			Basic: &graphql.BasicCredentialDataInput{
+				Username: "custom",
+				Password: "auth",
+			}},
+		AdditionalHeaders: &graphql.HttpHeaders{
+			"customHeader": []string{"custom", "custom"},
+		},
+	}
+
+	exampleSaved := false
+
+	testCases := []struct {
+		Name string
+		Apis []*graphql.APIDefinitionInput
+	}{
+		{
+			Name: "API without default auth",
+			Apis: []*graphql.APIDefinitionInput{
+				{
+					Name:      "without-default-auth",
+					TargetURL: "http://mywordpress.com/comments",
+				},
+			},
+		},
+		{
+			Name: "API with default auth",
+			Apis: []*graphql.APIDefinitionInput{
+				{
+					Name:        "with-default-auth",
+					TargetURL:   "http://mywordpress.com/comments",
+					DefaultAuth: defaultAuth,
+				},
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.Name, func(t *testing.T) {
+			appInput := graphql.ApplicationInput{
+				Name: "test-app",
+				Apis: testCase.Apis,
+				Labels: &graphql.Labels{
+					"scenarios": []interface{}{"DEFAULT"},
+				},
+			}
+
+			app := createApplicationFromInputWithinTenant(t, ctx, appInput, defaultTenant)
+			defer deleteApplication(t, app.ID)
+
+			var rtmIDs []string
+			for i := 0; i < 10; i++ {
+				rtm := createRuntime(t, ctx, fmt.Sprintf("test-rtm-%d", i))
+				rtmIDs = append(rtmIDs, rtm.ID)
+			}
+			defer func() {
+				for _, id := range rtmIDs {
+					deleteRuntime(t, id)
+				}
+			}()
+			require.True(t, len(rtmIDs) > 1)
+
+			rtmIDWithUnsetRtmAuth := rtmIDs[0]
+			rtmIDWithSetRtmAuth := rtmIDs[1]
+
+			setAPIAuth(t, ctx, app.Apis.Data[0].ID, rtmIDWithSetRtmAuth, customAuth)
+			defer deleteAPIAuth(t, ctx, app.Apis.Data[0].ID, rtmIDWithSetRtmAuth)
+
+			innerTestCases := []struct {
+				Name         string
+				QueriedRtmID string
+			}{
+				{
+					Name:         "Query set Runtime Auth",
+					QueriedRtmID: rtmIDWithSetRtmAuth,
+				},
+				{
+					Name:         "Query unset Runtime Auth",
+					QueriedRtmID: rtmIDWithUnsetRtmAuth,
+				},
+			}
+
+			for _, innerTestCase := range innerTestCases {
+				t.Run(innerTestCase.Name, func(t *testing.T) {
+					result := ApplicationExt{}
+
+					request := gcli.NewRequest(
```

Comment:
Create fixture for this request and then use `request.Query()` to save the content of the request.